st3/sublime_lib/fancy_settings.py

```diff
@@ -0,0 +1,47 @@
+class FancySettings():
+    def __init__(self, settings):
+        self.settings = settings
+
+    def get(self, name, default=None):
+        return self.settings.get(name, default)
+
+    def set(self, name, value):
+        self.settings.set(name, value)
+
+    def erase(self, name):
+        self.settings.erase(name)
+
+    def has(self, name):
+        return self.settings.has(name)
+
+    def __getitem__(self, key):
+        return self.get(key)
+
+    def __setitem__(self, key, value):
+        self.set(key, value)
+
+    def __delitem__(self, key):
+        self.erase(key)
+
+    def __contains__(self, item):
+        return self.has(item)
+
+    def subscribe(self, key, selector, callback):
+        if isinstance(selector, str):
+            key = selector
+            selector = lambda this: this[key]
+
+        previous_value = selector(self)
+
+        def onchange():
+            nonlocal previous_value
+            new_value = selector(self)
+
+            if new_value != previous_value:
+                callback(new_value, previous_value)
+                previous_value = new_value
+
+        self.settings.add_on_change(key, onchange)
```

Comment:
I prefer using `functools.partial` over closures in situations like this, because closures create a new function for each invocation, whereas partial just binds a parameter. I would define `_on_change` as a private method then.

Reply:
In this case, I think we need a closure (and a separate function instance per subscription) because we need to keep track of the last selected value.

Reply:
True, I missed that it was being modified for some reason. Yeah, I guess we need it then.